parcels/kernels/error.py

```diff
@@ -46,6 +47,17 @@ def __init__(self, particle, lon=None, lat=None, field=None):
         super(OutOfBoundsError, self).__init__(particle, msg=message)
 
 
+class TimeExtrapolationError(KernelError):
```

Comment:
Not strictly necessary, but how about `OutOfTimeError` instead of `TimeExtrapolationError` for consistency?